CMSgov/onemac
# OY2 5199 User profile - State User can self-revoke their access to a state

Story: https://qmacbis.atlassian.net/browse/OY2-5199
Endpoint: https://dymu0dlxo6b0e.cloudfront.net/
Figma: https://www.figma.com/file/OOtw6NirwsWqUlYkVW9pdz/Launch?node-id=11%3A11824
 
Changes:
- Cleanup: Updated the class to className in Footer.js, HomeFooter.js, and HomeHeader.js
- Cleanup: removed useCallback import from Signup.js
- Cleanup: updated UserManagement with new dependency and added callback
- UserPage - added the ‘x’ from the figma to the cards for State Users with pending or active status
- UserPage - added click event to ‘x’ to revoke access after browser confirmation
- UserPage - User details update with new status
- Added “ClosingX.svg” from the Figma
- Updated index.scss to float the ‘X’ into the proper place on the card
- Updated putUser -- added self-revoke email option to the admins
 
### Test Plan
Create your own users for State User and State Admin in the DynamoDB and use valid emails for testing the emails.  Start with them both "granted" access to the same state (VA is a good choice as it has a decent number of entries)

1. Log in as stateuserdenied@cms.hhs.local
2. Click on “My Account” and select “Manage Account”
3. Verify that the User Page contains a State Card with the “Denied” status and NO X
4. Logout
5. Log in as your state user test account - active with the state that your state admin is associated with...
6. Click on “My Account” and select “Manage Account”
7. Verify that the User Page contains a State Card in the "Active" status and there is an X in the corner (as seen in design prototype)
8. Click the 'X' and verify that the browser confirmation appears with the text:

>  "Warning Withdraw of State Access
> 
> This action cannot be undone. State User Admin will be notified. Are you sure you would like to withdraw State Access?
> Are you sure you want to proceed?
> 
> Cancel / Yes"

10. Click cancel to remove confirmation window and take no action.
11. Click the ‘X' again and select 'Yes’ from the confirmation window
12. Verify that the Status changes to “Access Revoked” on the correct card
13. Verify that the 'X' is gone from that card, but remains on any card still active or pending
14. Verify an email is sent to the State User with the content found here
15. Verify an email is sent to all active State Admins for the state chosen with the content here.
16. Logout
17. Change your user to pending status and verify you can revoke from pending as well
